src/pkgcheck/checks/codingstyle.py

```diff
@@ -79,6 +79,29 @@ def feed(self, pkg):
                     yield DeprecatedEapiCommand(name, line=call, lineno=lineno+1, eapi=pkg.eapi, pkg=pkg)
 
 
+class EendMissingArg(results.LineResult, results.Style):
```

Comment:
Should be a warning result, not style.

Reply:
I'm not clear on the distinction here. `eend` doesn't fail, but doesn't do much either (other than print `[ok]`). But I'll change it.

Reply:
It's warning here given it's illegal, from PMS:
>    Indicates that the process begun with an ebegin message has completed. Takes one fixed argument, which is a   numeric return code, and an optional message in all subsequent arguments. If the first argument is 0, prints a success indicator; otherwise, prints the message followed by a failure indicator. Returns its first argument as exit status.

The idea here being: it's not just poor taste or a bit dodgy, it's definitely _wrong_. Portage shouldn't even accept it and won't in the future.

Reply:
Oh, I wasn't aware that it's actually required. Makes way more sense as a warning then, thanks @thesamesam

Reply:
Personally I'd advise avoiding using portage as a standard for what is acceptable in many cases and look directly at the spec instead. :laughing:

Reply:
> Personally I'd advise avoiding using portage as a standard for what is acceptable 

Good point :smile:

Reply:
Well, once https://github.com/gentoo/gentoo/pull/25134 is merged, these should all be fixed, and portage can make it a hard error :tada:

Reply:
... or maybe not, that might break a bunch of overlays.

Reply:
That's not a huge problem 😉

Reply:
It just reinforces my viewpoint that all these kind of checks should be also done in a linting tool so overlay maintainers can easily scan and fix their issues without having to build every package in their tree while hoping the situation isn't conditionally triggered.